cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -255,15 +317,28 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 		if seen[uint64(start)] {
 			continue
 		}
+
+		// Only allow the creation of partitions with known GUIDs
+		// TODO:UC20: also provide a mechanism for MBR (RPi)
+		var ptype string
+		if encryptData && s.Role == gadget.SystemData {
+			// e8 - MBR type for LUKS encrypted partition
+			// CA7D7CCB-63ED-4C53-861C-1742536059CC - GPT partition type for encrypted partition
+			ptype = partitionType(ptable.Label, "e8,CA7D7CCB-63ED-4C53-861C-1742536059CC")
+		} else {
+			ptype = partitionType(ptable.Label, p.Type)
+		}
```

Comment:
we really should simply not do this without further thoughts, the Oher Data Partion GUID allows for LUKS anyway, there's probably code in gadget that would break on this behind the back the change

Reply:
I'll leave this with the original gadget partition type.